src/announcements/react.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+  useCallback,
+} from "react";
+import type { Announcement, AnnoucementsUserSettings, State } from "./types";
+import {
+  groupAnnouncements,
+  localizeAnnouncements,
+  filterAnnouncements,
+} from "./logic";
+import fetchApi from "./api";
+import { useMachine } from "@xstate/react";
+import { announcementMachine } from "./machine";
+
+type Props = {
+  children: React$Node,
+  onLoad: () => Promise<{
```

Comment:
I don't like the wording onLoad/onSave because it makes think it's just an event (like onClick) but actually it's the actual load/save